modyn/evaluator/internal/grpc/evaluator_grpc_servicer.py

```diff
@@ -125,31 +126,34 @@ def evaluate_model(self, request: EvaluateModelRequest, context: grpc.ServicerCo
                 evaluation_started=False, eval_aborted_reason=EvaluationAbortedReason.MODEL_NOT_EXIST_IN_STORAGE
             )
 
-        dataset_info = request.dataset_info
-        dataset_size_req = GetDatasetSizeRequest(
-            dataset_id=request.dataset_info.dataset_id,
-            start_timestamp=dataset_info.start_timestamp if dataset_info.HasField("start_timestamp") else None,
-            end_timestamp=dataset_info.end_timestamp if dataset_info.HasField("end_timestamp") else None,
-        )
-        dataset_size_response: GetDatasetSizeResponse = self._storage_stub.GetDatasetSize(dataset_size_req)
-
-        dataset_size = dataset_size_response.num_keys
-        if not dataset_size_response.success:
-            logger.error(
-                f"Total number of keys for dataset {dataset_size_req.dataset_id} cannot be fetched. "
-                f"Evaluation cannot be started."
-            )
-            return EvaluateModelResponse(
-                evaluation_started=False, eval_aborted_reason=EvaluationAbortedReason.DATASET_NOT_FOUND
+        data_sizes: list[int] = []
+        for interval in request.dataset_info.evaluation_intervals:
```

Comment:
Before I start reviewing, could you clarify what you mean by batching is done at the server? I am a little bit confused. I thought the client needs to group all eval requests by model id, and then submit a single request with all intervals for that model. How can you do this at the server? And this also looks like the server iterates over the intervals. I am a bit confused